spec/converter_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'spec_helper'
+require 'ronin/masscan/converter'
+require 'masscan/output_file'
+require 'tempfile'
+
+RSpec.describe Ronin::Masscan::Converter do
+  let(:fixtures_path) { File.expand_path(File.join(__dir__, '..', 'spec', 'fixtures')) }
+  let(:masscan_path)  { File.join(fixtures_path, 'converter', 'masscan.json') }
+  let(:masscan_file)  { Masscan::OutputFile.new(masscan_path) }
+  let(:expected_json) do
+    "[{\"status\":\"open\",\"protocol\":\"tcp\",\"port\":80,\"reason\":[\"syn\",\"ack\"],\"ttl\":54,\"ip\":\"93.184.216.34\",\"timestamp\":\"2021-08-26 08:50:21 +0000\"}]"
+  end
+
+  describe '.convert_file' do
+    it 'must convert masscan file and wirte it into a file' do
+      Tempfile.create(['dest', '.json']) do |output_file|
+        subject.convert_file(masscan_path, output_file)
+        output_file.rewind
+
+        expect(output_file.read).to eq(expected_json)
+      end
+    end
+  end
+
+  describe '.convert' do
+    context 'when there is no output' do
+      context 'and format is csv' do
+        let(:expected_csv) do
+          "type,status.status,status.protocol,status.port,status.reason,status.ttl,status.ip,status.timestamp,banner.protocol,banner.port,banner.ip,banner.timestamp,banner.app_protocol,banner.payload\nstatus,open,tcp,80,\"syn,ack\",54,93.184.216.34,2021-08-26 08:50:21 +0200\n"
```

Comment:
Is there a way to make it multi-line but without adding some unnecessary `\n`'s?

Reply:
Do you mean the last newline? You can do either:

```ruby
<<~CSV.chomp
  foo,bar,baz
  ...
CSV
```

or

```ruby
[
  "foo,bar,baz",
  "..."
].join("\n")
```

Reply:
Not the last one. I don't want this `expected_csv` to be too long but saving it as `<<CSV.chomp ...` fails when I want to compare it with result of `.convert`